### PR TITLE
Makes SQLQuery more robust.

### PR DESCRIPTION
**Pull request by andyHa**
- Improves error reporting for malformed prepared statement parameters.
- Warns if a SQL query references an unknown variable.

### CODE REVIEW

**Review by meg1502 — APPROVED** [review]

File: src/main/java/sirius/db/jdbc/StatementCompiler.java [hunk 1]
```diff
@@ -176,6 +192,17 @@ private Object computeEffectiveParameterValue(String fullParameterName) throws S
 
         Object paramValue = context.get(parameterName);
         if (accessPath == null || paramValue == null) {
+            if (!context.containsKey(parameterName)) {
+                Databases.LOG.WARN("""
+                                           SQL query references an unknown paramter: %s! \
+                                           This is very likely an error and thus reported. \
+                                           Set unknown parameters to null to suppress this warning!
+                                           Query: %s
+                                           Location: %s""",
```
Comment (meg1502) [inline src/main/java/sirius/db/jdbc/StatementCompiler.java:201]: Is the triple quotes + empty line intended?

**Review by cschierle — APPROVED** [review]

File: src/main/java/sirius/db/jdbc/StatementCompiler.java [hunk 2]
```diff
@@ -191,6 +192,17 @@ private Object computeEffectiveParameterValue(String fullParameterName) throws S
 
         Object paramValue = context.get(parameterName);
         if (accessPath == null || paramValue == null) {
+            if (!context.containsKey(parameterName)) {
+                Databases.LOG.WARN("""
+                                           SQL query references an unknown paramter: %s! \
```
Comment (idlira) [inline src/main/java/sirius/db/jdbc/StatementCompiler.java:197]: paramter -> parameter

File: src/main/java/sirius/db/jdbc/StatementCompiler.java (hunk 1 above)
Comment (sabieber) [reply src/main/java/sirius/db/jdbc/StatementCompiler.java:201]: Thats the new java multiline text syntax

**Review by idlira — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/db/jdbc/StatementCompiler.java (hunk 1 above)
Comment (meg1502) [reply src/main/java/sirius/db/jdbc/StatementCompiler.java:201]: yes they are - https://openjdk.java.net/jeps/355 for anyone else wondering

Comment (sabieber) [reply src/main/java/sirius/db/jdbc/StatementCompiler.java:201]: https://www.baeldung.com/java-multiline-string

**Review by sabieber — APPROVED** [review]

**Review by fhaScireum — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

File: src/main/java/sirius/db/jdbc/StatementCompiler.java (hunk 1 above)
Comment (andyHa) [reply src/main/java/sirius/db/jdbc/StatementCompiler.java]: Yep, this is a multiline string literal and yields a message like:
```
[2021-06-17T17:20:51.317] WARNING [jdbc - StatementCompiler.java:196] SQL query references an unknown paramter: filter! This is very likely an error and thus reported. Set unknown parameters to null to suppress this warning!
Query: SELECT * FROM test_a [WHERE a = ${filter}]
Location:
Execution Point in: main
---------------------------------------------------
sirius.kernel.async.ExecutionPoint.snapshot (ExecutionPoint.java:59)
sirius.kernel.async.ExecutionPoint.snapshot (ExecutionPoint.java:42)
sirius.db.jdbc.StatementCompiler.computeEffectiveParameterValue (StatementCompiler.java:205)
sirius.db.jdbc.StatementCompiler.compileSectionPart (StatementCompiler.java:167)
...
...
```

**Pull request merged**